src/controllers/EventController.ts

```diff
@@ -105,8 +117,46 @@ export class EventController {
     return this.eventMapper.entityToResponse(deletedEvent);
   }
 
+  @Get('/:eventID/attendance')
+  @ResponseSchema(MultipleAttendanceResponse)
+  @UseBefore(OfficerAuthMiddleware)
+  @OpenAPI({ security: [{ TokenAuth: [] }] })
+  async getEventAttendance(
+    @Param('eventID') eventID: number,
+    @QueryParams() multipleAttendanceQuery: MultipleAttendanceQuery
+  ): Promise<MultipleAttendanceResponse | undefined> {
+    const attendances = await this.eventService.getAttendancesFromEvent(
+      eventID,
+      multipleAttendanceQuery
+    );
+
+    return attendances ? { attendances } : undefined;
+  }
+
+  @Post('/:eventID/attendance')
+  @ResponseSchema(AttendanceResponse)
+  @UseBefore(OfficerAuthMiddleware)
+  @OpenAPI({ security: [{ TokenAuth: [] }] })
+  async checkOffEventAttendance(
+    @Param('eventID') eventID: number,
+    @Body() attendanceCheckOffRequest: AttendanceCheckOffRequest
+  ): Promise<AttendanceResponse | undefined> {
+    const { attendeeId, officerId, ...otherValues } = attendanceCheckOffRequest;
+
+    if (officerId == undefined) {
+      return undefined;
+    }
+
+    const [event, attendee, officer] = await Promise.all([
```

Comment:
we don't actually need to query the objects, we should be able to save just with the id
i think i did it with the events where we had hosts as the PKpayload

Reply:
I think this can only be done for officer column, since event and attendee columns are primary columns @godwinpang

Reply:
nvm I made it work